src/wcmValidateDevice.c

```diff
@@ -204,9 +204,14 @@ int wcmDeviceTypeKeys(InputInfoPtr pInfo)
 
 		case 0x34D: /* MobileStudio Pro 13 */
 		case 0x34E: /* MobileStudio Pro 16 */
+		case 0x390: /* Cintiq 16 */
+		case 0x391: /* Cintiq 22 */
+		case 0x396: /* DTK-1660E */
 		case 0x398: /* MobileStudio Pro 13 */
 		case 0x399: /* MobileStudio Pro 16 */
 		case 0x3AA: /* MobileStudio Pro 16 */
+		case 0x3AE: /* Cintiq 16 */
+		case 0x3B0: /* DTK-1660E */
 			TabletSetFeature(priv->common, WCM_LCD);
```

Comment:
This doesn't look like the right spot to add these cases. The fall-through will cause these devices to get the `WCM_LCD`, `WCM_ROTATION`, and `WCM_RING` features, but I don't believe these devices have a touchring.